nsv13/code/modules/overmap/traders_items.dm

```diff
@@ -25,9 +28,123 @@
 		SEND_SOUND(user, 'nsv13/sound/effects/ship/freespace2/computer/textdraw.wav')
 		to_chat(user, "<span class='boldnotice'>[pick(inhabited_trader.greetings)]</span>")
 
-//Nope!
+/obj/structure/overmap/trader/proc/add_objective( objective )
+	if ( objective )
+		expecting_cargo += objective 
+		essential = TRUE
+
+/obj/structure/overmap/trader/receive_cargo( mob/living/user, var/obj/machinery/computer/ship/dradis/cargo/console, var/obj/item/ship_weapon/ammunition/torpedo/freight/shipment )
+	if ( inhabited_trader )
+		var/datum/freight_delivery_receipt/receipt = new /datum/freight_delivery_receipt()
+		receipt.courier = user
+		receipt.vessel = console.linked 
+		receipt.shipment = shipment
+		receipts += receipt
+
+		check_objectives( receipt )
+	else 
+		. = ..() // Fallback to attempting a physical delivery to a station zlevel, if it exists 
+	
+/obj/structure/overmap/trader/proc/check_objectives( var/datum/freight_delivery_receipt/receipt )
+	if ( !length( expecting_cargo ) ) 
+		if ( receipt && receipt.vessel )
+			SEND_SOUND(receipt.courier, 'nsv13/sound/effects/ship/freespace2/computer/textdraw.wav')
+			receipt.vessel.hail( pick( list( 
+				"We're not expecting any shipments at this time. Please give us some time to arrange the return shipment.",
+				"We're not expecting any shipments, please don't send us your trash.",
+				"This cargo isn't registered on our supply requests. We will return it as soon as we can.",
+				"We haven't asked for any cargo like this. Take your business elsewhere.",
+			) ), src)
+			addtimer(CALLBACK(src, .proc/return_shipment, receipt), 60 SECONDS)
+		return FALSE 
+	
+	for ( var/request in expecting_cargo ) 
+		if ( istype( request, /datum/overmap_objective/cargo ) )
+			var/datum/overmap_objective/cargo/objective = request 
+			var/success = objective.check_cargo( receipt.shipment ) 
+			
+			if ( success ) 
+				if ( receipt && receipt.vessel )
+					SEND_SOUND(receipt.courier, 'nsv13/sound/effects/ship/freespace2/computer/textdraw.wav')
+					receipt.vessel.hail( "Thank you for delivering this cargo. We have marked the supply request as received.", src)
+				// Bag it, tag it, store it. Accessible for admin debugging later if needed 
+				receipt.completed_objective = objective 
+				received_cargo += receipt
+				addtimer(CALLBACK(src, .proc/return_approved_form, receipt), 60 SECONDS)
```

Comment:
return after this addtimer if successful. then take the next block and move it out of the else, and remove the duplicate block from the top.